core/src/main/resources/lib/hudson/newFromList/validation.js

```diff
@@ -25,4 +25,40 @@ function updateOk() {
     document.getElementById("ok").disabled = state();
 }
 
-updateOk();
\ No newline at end of file
+updateOk();
+
+document.addEventListener('DOMContentLoaded', function() {
+  let nameField = document.getElementById('name');
+  nameField.focus();
+  nameField.addEventListener('change', function () {
+    updateOk();
+  });
+  nameField.addEventListener('keyup', function () {
+    updateOk();
+  });
+
+  document.querySelectorAll('.mode-selection').forEach(function(el) {
+    el.addEventListener('change', function () {
+      updateOk();
+    });
+    el.addEventListener('click', function () {
+      updateOk();
+    });
+  });
+
+  let copyRadio = document.getElementById('copy');
+  if (copyRadio !== null) {
+    copyRadio.addEventListener('change', function () {
+      updateOk();
+    });
+    copyRadio.addEventListener('keyup', function () {
```

Comment:
Well spotted! This went through a few iterations while I figured out what the new behavior should be, since the text field is now hidden by default (so the old code didn't even work). So this started life as events on the (hidden) text field.